File: solutions/882.peak-index-in-a-mountain-array/peak-index-in-a-mountain-array.java
class Solution {
    public int peakIndexInMountainArray(int[] A) {
        int i = 1;
        while(i < A.length) {
            if(A[i] < A[i - 1]) return i - 1;
            i++;
        }
        return 0;
    }
}

File: solutions/882.peak-index-in-a-mountain-array/peak-index-in-a-mountain-array.cpp
class Solution {
public:
    int peakIndexInMountainArray(vector<int>& A) {
        int left = 0, right = A.size() - 1;
        int mid = (left + right) / 2;
        while(left < right) {
            if(A[mid] < A[mid + 1]) left = mid + 1;
            else right = mid;
            mid = (left + right) / 2;
        }
        return left;
    }
};